packages/service/lib/bus.js

```diff
@@ -0,0 +1,40 @@
+const { EventEmitter } = require('events')
+
+class Bus extends EventEmitter {
+  #id
+  #channel
+
+  constructor (id) {
+    super()
+
+    this.#id = id
+    this.#channel = new BroadcastChannel('plt:bus')
+    this.#channel.onmessage = this.#handleMessage.bind(this)
+    this.#channel.unref()
+  }
+
+  close () {
+    this.#channel.close()
+  }
+
+  send (destination, type, data) {
+    this.#channel.postMessage({ source: this.#id, destination, type, data })
+  }
+
+  broadcast (type, data) {
+    this.send('*', type, data)
+  }
+
+  #handleMessage ({ data: message }) {
+    const { type, destination } = message
+
+    if (destination !== '*' && destination !== this.#id) {
+      return
+    }
+
+    this.emit('message', message)
+    this.emit(type, message)
```

Comment:
This will cause a process/thread crash if `type` is `'error'`. `error` should not allowed as a type.

Reply:
Done!